### ISSUE TITLE
Missing permissions to Graph endpoint (mismatch of endpoint with Azure AD API permissions)

### ISSUE THREAD

**Issue by MMollyy** [Azure/azure-sdk-for-python#6401]
Hi,

I would like to report a mismatch of the permissions being assigned from Azure AD to the credential which is used in Python to perform actions against the graph.windows.net endpoint.

Scenario sketch:
- Application/SPN is created in Azure AD
- API permissions are assigned on this
- Using azure-graphrbac library: ServicePrincipalCredentials() and GraphRbacManagementClient() is being used to authenticate against Graph and perform actions on Applications, Users, Groups and Roles.

Right now I have API permissions assigned and I am able to authenticate, but have insufficient permissions to perform any action on applications, users or groups.
The API permissions assigned to the application are:
- Microsoft Graph: (Application permissions)
 - - Application.ReadWrite.All
 - - Group.ReadWrite.All
 - - RoleManagement.ReadWrite.Directory
 - - Users.Invite.All
 - - Users.ReadWrite.All

Using the permissions for Microsoft Graph doesn't work, while this should be the one we assign permissions for according to: https://github.com/MicrosoftDocs/azure-docs-sdk-python/blob/master/docs-ref-services/graph-rbac.md
I quote:
> [!NOTE] You need to change the resource parameter to https://graph.windows.net while creating the credentials instance <
```graph_credentials = ServicePrincipalCredentials(client_id=AZURE_CLIENT_ID, secret=AZURE_CLIENT_SECRET,
                                                tenant=AZURE_TENANT_ID,
                                                resource=AZURE_PUBLIC_CLOUD.endpoints.active_directory_graph_resource_id
                                                )
graph_client = GraphRbacManagementClient(graph_credentials, AZURE_TENANT_ID, base_url=None)
```

However, since that didn't work I decided to switch to the API Permissions for supported **Legacy APIs**: 

- Azure Active Directory Graph: (Application permissions)

- - Application.ReadWrite.All

Note here that there are no permissions for Users, Groups or Roles. Which is a different issue from this ticket. But I wouldn't know where to report that at this moment.
However using this legacy API permissions, and accessing this endpoint, I am able to perform actions on applications and service principals.
I also tested the other possible endpoints with the Microsoft Graph API permissions, but none of them gave me a working scenario.

So there are a few things here:
Should there be some other endpoint that we use so that the Microsoft Graph API permissions actually work?
Maybe not, then I would kindly ask the product group to be contacted to make sure that the Microsoft Graph API permissions work on the endpoint? And not just the Legacy API permissions.
Maybe I am using an incorrect library,  and the endpoint is incorrect as well?

**Comment by loarabia** [comment]
Hi @MMollyy , thanks for taking the time to report the inconsistency. I've labeled the bug to route it to the right team.

**Comment by MMollyy** [comment]
thanks @loarabia I'll be looking forward to the response.

**Comment by gokboru** [comment]
We have the same problem.

**Comment by lilyjma** [comment]
@MMollyy and @gokboru : are you still experiencing the same problem?

**Comment by MMollyy** [comment]
Well, I tested the legacy AP permissions. It seems to have all the other permissions, but it doesn't allow me to delete groups. Even though Groups.ReadWrite.All is enabled. Kind of moot point when it's legacy though. Deleting actually also did not work when changing to the https://graph.windows.net endpoint - I had to change my code from using the Azure SDK to using the request library to make an actual REST API call from the code towards the Graph API. And then it worked.
The rest worked on the Legacy API with the SDK.

I've yet to test the application registration & service principal parts against the Graph API. I did not have time for it. 
I'm gonna skip on testing the user part for the Graph API with the SDK.
